src/eval.ml

```diff
@@ -34,6 +37,23 @@ let to_term : term -> stack -> term = fun t args ->
     | u::args -> to_term (Appl(t,snd Pervasives.(!u))) args
   in to_term t args
 
+(** [extract x p] extracts the term at position [p] from list of terms [x]. *)
+let extract : term list -> Position.t -> term = fun xs p ->
+  let rec loop xs p acc = match p with
+    | [] -> acc
+    | hdp :: tlp ->
+       let elt = List.nth xs hdp in
```

Comment:
List.nth can fail. You should do something about this.

Reply:
this function is useless and has been removed in 9661fdf